tool/matomo/classes/tool/tool.php

```diff
@@ -153,35 +160,150 @@ public function form_validate(&$data, &$files, &$errors) {
     /**
      * Build settings array from submitted form data.
      *
-     * @param \stdClass $data
+     * @param stdClass $data
      *
      * @return array
      */
-    public function form_build_settings(\stdClass $data): array {
+    public function form_build_settings(stdClass $data): array {
         $settings = [];
-        $settings['siteid']  = isset($data->siteid) ? $data->siteid : '';
+        $settings['siteid'] = isset($data->siteid) ? $data->siteid : '';
```

Comment:
this is a bit repetitive to me, can this be a $defaults array and then cast the stdClass to an array and merge them and skip all the isset bits?